### PR TITLE
Groups

### PR DESCRIPTION
**Pull request by YaremaKryven**
Added models and forms for Groups, Shedule and Invitation

### CODE REVIEW

File: groups/validators.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+def dates_validator(self, start, end, not_in_past, name, field, items=1):
+    if not items:
+        msg = u'Input number of students to invite to this group.'
+        self.add_error('items_left', msg)
+    if not_in_past < timezone.now():
+        msg = name.title() + u' date already passed! Please enter valid date.'
+        self.add_error(field, msg)
+    if start >= end:
+        msg = u'End date should be after start date!'
+        self.add_error('end', msg)
```
Comment (MrKreg) [inline groups/validators.py]: Using method add_error is useless for serializers.

File: groups/validators.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from django.utils import timezone
+
+
+def dates_validator(self, start, end, not_in_past, name, field, items=1):
```
Comment (MrKreg) [inline groups/validators.py]: It’s not a method, which “self” are you meaning?

Comment (falled10) [inline groups/validators.py]: 1. There is too many local variables, can you remove some variables?
2. What does self variable mean in your method?
3. Why your items variable by default equal 1?
4. What start, end, name variables mean?
5. Why your method returns nothing, maybe you can return cleaned_data here?

**Review by falled10 — CHANGES_REQUESTED** [review]

File: groups/validators.py (hunk 2 above)
Comment (MrKreg) [reply groups/validators.py]: It will be better to use only 1 parameter “data” that contains all data.

**Review by MrKreg — CHANGES_REQUESTED** [review]

**Pull request closed without merging**